mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1199,9 +1239,9 @@ private static AssociationStorageStrategy getAssociationStorageStrategy(Associat
 	public void executeBatch(OperationsQueue queue) {
 		if ( !queue.isClosed() ) {
 			Operation operation = queue.poll();
-			Map<DBCollection, BatchInsertionTask> inserts = new HashMap<DBCollection, BatchInsertionTask>();
+			Map<MongoCollection<Document>, BatchInsertionTask> inserts = new HashMap<MongoCollection<Document>, BatchInsertionTask>();
 
-			List<Tuple> insertTuples = new ArrayList<Tuple>();
+			List<Tuple> insertTuples = new LinkedList<Tuple>();
```

Comment:
hi @Sanne !

I will restore ArrayList. it is not a problem. 
Broken formatting .... yes You right.  Easy to restart more carefully.

About migration to 3.4 api ... It is large and complex change. And ... of course I can analyse and do self. But ... I hope to help of members because they are authors.